### ISSUE TITLE
Update reference scripts to use the "Batteries Included" utils

### ISSUE THREAD

**Issue by datumbox** [pytorch/vision#4281]
## 🚀 Feature
As part of the "Batteries Included" initiative (#3911) we are adding a number of new utils that can be used to produce SOTA results. Once those utils are landed, we should update our reference scripts to use them. 

More specifically we need to:
- [ ] Update the reference scripts to use PyTorch's new [linear warmup](https://github.com/pytorch/pytorch/pull/60836) scheduler.
  - The reference scripts should be updated to let users define the `warmup_method` (if any), the `warmup_iters` and the `warmup_factor`. The warmup scheduler should be chained with other existing schedulers.
  - The Object Detection recipe needs to be BC compatible and replace our [custom linear warmup](https://github.com/pytorch/vision/blob/dc42f933f3343c76727dbfba6e4242f1bcb8e1a0/references/detection/engine.py#L20-L24) approach with the one from PyTorch.
  - Similarly the Video Classification recipe needs also to be BC compatible and replace our [customer scheduler](https://github.com/pytorch/vision/blob/master/references/video_classification/scheduler.py#L5).
  - All other recipes (Classification and Segmentation), should be updated to optionally use warm up. The addition should be BC compatible and turned off by default.
- [x] Update the Classification reference to use PyTorch's new label smoothing implementation.
  - We will support the case of label smoothing by using the upcoming `label_smoothing` argument of CrossEntropyLoss (https://github.com/pytorch/pytorch/pull/63122). 
  - A new `label_smoothing` argument should be introduced on the [script](https://github.com/pytorch/vision/blob/c2ab0c59f42babf9ad01aa616cd8a901daac86dd/references/classification/train.py#L165). To ensure BC, its default value should be 0.0 (no label smoothing).
